### PR TITLE
Refactor storage service to improve modularity and error handling

### PR DESCRIPTION
**Pull request by ashleythedeveloper**
## Description
This PR refactors the storage service to improve modularity, error handling, and type safety. It also adds new functionality for document storage and updates the cryptography service.

## What type of PR is this? (check all applicable)
- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
N/A

## Added tests?
- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?
- [x] 📜 README.md
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/services/cryptography/crypto.ts [hunk 1]
```diff
@@ -1,46 +1,65 @@
...
+     * @returns {IEncryptionResult} An object containing the encrypted data:
+     *          - cipherText: The encrypted string (Base64 encoded).
+     *          - iv: The initialization vector used for encryption (Base64 encoded).
+     *          - tag: The authentication tag for GCM mode (Base64 encoded).
+     *          - type: The encryption algorithm used.
+     * @throws {Error} If the key length is invalid for the specified algorithm.
      */
-    encryptString(input: string, key: string) {
-        const encryptedDocument = oaEncryptString(input, key);
+    encryptString(input: string, key: string, algorithm = EncryptionAlgorithm.AES_256_GCM) {
```
Comment (ldhyen99) [inline src/services/cryptography/crypto.ts:43]: I just curious about this function. Why don't we use the library like `@govtechsg/oa-encryption` for encryption?

File: README.md [hunk 2]
```diff
@@ -1,88 +1,187 @@
...
 # Install dependencies
 yarn install
 
-# Run the app
-yarn start
+# Run the app and watch for changes
+yarn dev
 
 # Build the app
 yarn build
```
Comment (huynguyen-hl) [inline README.md:45]: I believe the `yarn build` step should be placed before the `yarn dev` step, as the `yarn dev` step triggers `watch:server`, which relies on the `dist` folder.

**Review by namhoang1604 — APPROVED** [review]
LGTM

**Review by ldhyen99 — APPROVED** [review]

File: src/services/cryptography/crypto.ts (hunk 1 above)
Comment (ashleythedeveloper) [reply src/services/cryptography/crypto.ts:43]: @ldhyen99, this is a good question. 

The main reason is to decouple the encryption method from the way in which Open Attestation does it.

For example, in the current implementation, `@govtechsg/oa-encryption` uses an OPEN-ATTESTATION-TYPE-1. Also, what happens if OA no longer wants to use the AES-GMC algorithm?

File: README.md (hunk 2 above)
Comment (ashleythedeveloper) [reply README.md:45]: @huynguyen-hl, good catch.

**Review by ldhyen99 — APPROVED** [review]

**Review by huynguyen-hl — APPROVED** [review]
LGTM

**Pull request merged**